planner/core/find_best_task.go

```diff
@@ -43,6 +43,10 @@ const (
 	// of a Selection or a JoinCondition, we can use this default value.
 	SelectionFactor = 0.8
 	distinctFactor  = 0.8
+
+	// When scan row count is small, the performance gap between asc and
```

Comment:
The difference is not big is not a good reason to make a change.
I think we should explain it with a solid reason like:
```
If the actual row count is much more than the limit count, the unordered scan may cost much more than keep order.
So when a limit exists, we don't apply the DescScanFactor.
```